src/components/dialog.js

```diff
@@ -230,15 +233,45 @@ export class Dialog {
   }
 
   /**
-   * Whether to display loading indicator.
-   * @param {boolean} isLoading
+   * Transitions to the next view
+   * @param {!./view.View} view
    */
-  setLoading(isLoading) {
-    if (isLoading) {
+  entryTransitionToNextView(view) {
+    // Temporarily cache the old view
+    this.previousView_ = this.view_;
+    var noloadingIndicatorPresent = !this.previousView_
+        || (this.previousView_
+            && !this.previousView_.hasLoadingIndicator());
+    // If previous view did not display loading indicator,
+    // remove contents of old view if present
+    if (noloadingIndicatorPresent) {
+      removeChildren(this.getContainer());
+    }
+    setImportantStyles(view.getElement(), resetViewStyles);
+    if (noloadingIndicatorPresent) {
+      // When loading indicator was not displayed in the previous view,
+      // loading indicator must be displayed while transitioning to new view.      
       this.loadingView_.show();
-    } else {
+      // Keep track of whether or not a view shows loading indicator
+      view.setLoadingIndicator(true);   
+    }
+  }
+
+   /**
+   * Transition out of an old view
+   * @param {boolean} hideLoadingView 
+   */
+  exitTransitionFromOldView(hideLoadingView) {
+    if (hideLoadingView) {
+      //When a loading indicator is present, hide it.
       this.loadingView_.hide();
     }
+    // If previous view displayed a loading indicator, this view
+    // did not, hence remove previous element instead.
+    if (this.previousView_ && this.previousView_.hasLoadingIndicator()) {
```

Comment:
Ditto, if `this.previousView_` exists and attached, should we just always remove it? It doesn't matter whether `hasLoadinfgIndicator` is true, right?

E.g.

if (this.previousView_ && this.previousView_.getElement().parentNode) {
  removeElement(this.previousView_.getElement())
}

Reply:
done